src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -3491,6 +3492,13 @@ def _update_addons(cmd, instance, subscription_id, resource_group_name, name, ad
                     enabled=True, config={CONST_ACC_SGX_QUOTE_HELPER_ENABLED: "false"})
                 if enable_sgxquotehelper:
                     addon_profile.config[CONST_ACC_SGX_QUOTE_HELPER_ENABLED] = "true"
+            elif addon == CONST_OPEN_SERVICE_MESH_ADDON_NAME:
+                if addon_profile.enabled:
+                    raise CLIError('The open-service-mesh addon is already enabled for this managed cluster.\n'
```

Comment:
Could you please use an specific error type instead of `CLIError`. For more details, please refer to the doc [error_handling_guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md)

Reply:
@zhoxing-ms I was being consistent with similar errors raised in this function. I don't mind deviating from it though.

Reply:
OK, please improve it. The previous code is not standardized. I suggest you consider refactoring them in the future

Reply:
I have updated this.